### PR TITLE
[Merged by Bors] - TY-2471 test requestFeed

### PR DESCRIPTION
**Pull request by Robert-Steiner**
Ticket: 
- [TY-2471]

[TY-2471]: https://xainag.atlassian.net/browse/TY-2471?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: discovery_engine/test/integration/request_feed_test.dart [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        'requestNextFeed', () async {
+      server = await LocalNewsApiServer.start();
+      final engine = await DiscoveryEngine.init(
+        configuration: createConfig(data, server.port),
+      );
+
+      final nextBatchResponse = await engine.requestNextFeedBatch();
+      final restoreFeedResponse = await engine.requestFeed();
+
+      expect(restoreFeedResponse, isA<FeedRequestSucceeded>());
```
Comment (janpetschexain) [inline discovery_engine/test/integration/request_feed_test.dart:57]: shouldn't we also type check the `nextBatchResponse` before casting it?
```suggestion
      expect(nextBatchResponse, isA<NextFeedBatchRequestSucceeded>());
      expect(restoreFeedResponse, isA<FeedRequestSucceeded>());
```

File: discovery_engine/test/integration/request_feed_test.dart [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    });
+
+    tearDown(() async {
+      await server.close();
+      await Directory(data.applicationDirectoryPath).delete(recursive: true);
+    });
+
+    test(
+        'requestFeed should return the feed that has been requested before with'
+        'requestNextFeed', () async {
```
Comment (janpetschexain) [inline discovery_engine/test/integration/request_feed_test.dart:48]:
```suggestion
        'requestNextFeedBatch', () async {
```

File: discovery_engine/test/integration/request_feed_test.dart [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+      expect(restoreFeedResponse, isA<FeedRequestSucceeded>());
+      expect(
+        (nextBatchResponse as NextFeedBatchRequestSucceeded).items,
+        equals((restoreFeedResponse as FeedRequestSucceeded).items),
+      );
+    });
+
+    test(
+        'if requestNextFeed fails due to a news api request error, requestFeed'
```
Comment (janpetschexain) [inline discovery_engine/test/integration/request_feed_test.dart:65]:
```suggestion
        'if requestNextFeedBatch fails due to a news api request error, requestFeed'
```

**Review by janpetschexain — APPROVED** [review]
lgtm

**Comment by Robert-Steiner** [comment]
bors merge

**Pull request closed without merging**